### ISSUE TITLE
Differences in coordinate handling between backend and frontend

### ISSUE THREAD

**Issue by potatoeggy** [potatoeggy/connectfour#20]
Graphical:  
```
0 0 0 0 0 0 
0 0 0 0 0 0 
0 0 0 0 0 0 
1 1 1 1 1 0 
0 0 0 0 0 0 
0 0 0 0 0 0 
```
Backend:  
```
0 0 0 0 0 1 0
 0 0 0 0 0 1 0
 0 0 0 0 0 1 0
 0 0 0 0 0 1 0
 0 0 0 0 0 1 0
 0 0 0 0 0 0 0
```
The graphical board uses the first dimension for rows and the second for columns, then passes the value of the second dimension (columns) to addChip. The board then sets the square at coordinate (addChip, column) to a colour.

@bob2957, can you provide documentation for how the board stores its data?

**Comment by vincentjguo** [comment]
For the backend the first dimension is for row and the second is for column. However it flips the row number because for an array 0 is the top.

**Comment by potatoeggy** [comment]
Oops, I was using Arrays.deepToString to determine the layout of the backend board. Here are actual results:

Graphical:
```
0 0 0 0 0 0
0 0 0 0 0 0 
0 0 0 0 0 0 
0 0 0 0 0 0 
0 1 1 1 1 0 
0 0 0 0 0 0 
0 0 0 0 0 0 
```
Backend:
```
0 0 0 0 0 0 0 
0 0 0 0 0 1 0 
0 0 0 0 0 1 0 
0 0 0 0 0 1 0 
0 0 0 0 0 1 0 
0 0 0 0 0 1 0 
```

I've noticed that the frontend board is 7 x 6 instead of 6 x 7. That's probably the problem.

**Comment by vincentjguo** [comment]
by the way I have a final variable in the Board class that you can use to change the dimensions of the board. It will affect everything.

**Comment by potatoeggy** [comment]
Because of GridLayout issues, this must be reopened until I can find a way around this, and will need a major graphical rewrite.

Current backend layout, please correct me if I'm wrong:
```
 0 1 2 3 4 5 6
0
1
2
3
4
5
```

**Comment by vincentjguo** [comment]
yep that's it

**Comment by potatoeggy** [comment]
wait i'm confused now
![image](https://user-images.githubusercontent.com/25178974/83357691-db8bed80-a33b-11ea-9376-741595cb0e99.png)

Pressing anything in columns six (e.g., 1, 6) crashes

**Comment by potatoeggy** [comment]
With array out of bounds

**Comment by vincentjguo** [comment]
maybe fixed?

**Comment by potatoeggy** [comment]
Passing 6 to addChip (which should add a chip in column 6, zero-indexed) results in out of bounds.

**Comment by vincentjguo** [comment]
wait what branch are you running?

**Comment by potatoeggy** [comment]
i'm using a debug branch that has unpushed changes `front_back_connect` and master

**Comment by vincentjguo** [comment]
yep i pushed an update to that branch
make sure to update

**Comment by potatoeggy** [comment]
Fixed by `51083f6`.
